packages/block-editor/src/hooks/layout.js

```diff
@@ -56,6 +56,15 @@ function LayoutPanel( { setAttributes, attributes, name: blockName } ) {
 		return null;
 	}
 
+	// Only show the inherit toggle if it's supported,
+	// a default theme layout is set (e.g. one that provides `contentSize` and/or `wideSize` values),
+	// and that the default / flow layout type is in use, as this is the only one that supports inheritance.
+	const showInheritToggle = !! (
+		allowInheriting &&
+		!! defaultThemeLayout &&
+		( ! layout?.type || layout?.type === 'default' || layout?.inherit )
```

Comment:
Note: I've added `layout?.inherit` to this conditional so that any blocks that are currently in the weird state of having a `flex` type but inheriting from the default layout can switch off inherit via the UI.

Reply:
should we make this `( ! layout?.type || layout?.type === 'default' || (layout?.inherit  && layout?.type === 'flex') )`
so the toggle option disappears if they do toggle it off, or is that too confusing for the user to have it there and then it disappears?

Reply:
Should we make the toggle disappear if they do toggle it off in this state or is that too confusing for the user?

Reply:
Thanks for taking a look @glendaviesnz — currently, toggling `inherit` also switches the `type` to `default` since the `layout` attribute is cleared out of anything but the `inherit` property (so the toggle would still be visible). Given how the toggle works, I'm hoping we don't actually have anyone in that weird circumstance of `inherit` being set to `true` and the `type` being set to something other than `default`.

I mostly added the `layout?.inherit` to be cautious just in case something was set manually to that state, but I'm not sure if it ever would have been possible to get into that state via the UI, so it's possibly an artificial example 🤔